drivers/can/can_mcp2515.c

```diff
@@ -290,42 +290,45 @@ static int mcp2515_get_mode(const struct device *dev, uint8_t *mode)
 	return 0;
 }
 
-static int mcp2515_configure(const struct device *dev, enum can_mode mode,
-			     uint32_t bitrate)
+static uint32_t mcp2515_get_core_clock(const struct device *dev)
 {
 	const struct mcp2515_config *dev_cfg = DEV_CFG(dev);
+
+	return dev_cfg->osc_freq / 2;
+}
+
+
+static int mcp2515_configure(const struct device *dev, enum can_mode mode,
+			     const struct can_timing *timing,
+			     const struct can_timing *timing_data)
+{
+	ARG_UNUSED(timing_data);
 	struct mcp2515_data *dev_data = DEV_DATA(dev);
 	int ret;
 
 	/* CNF3, CNF2, CNF1, CANINTE */
 	uint8_t config_buf[4];
 	uint8_t reset_mode;
 
-	if (bitrate == 0) {
-		bitrate = dev_cfg->bus_speed;
-	}
-
-	const uint8_t bit_length = 1 + dev_cfg->tq_prop + dev_cfg->tq_bs1 +
-				dev_cfg->tq_bs2;
 
 	/* CNF1; SJW<7:6> | BRP<5:0> */
-	uint8_t brp = (dev_cfg->osc_freq / (bit_length * bitrate * 2)) - 1;
-	const uint8_t sjw = (dev_cfg->tq_sjw - 1) << 6;
+	uint8_t brp = timing->prescaler;
```

Comment:
Whoops, need to check for NULL.

Reply:
Just in case, this applies to all existing drivers :-)